lib/Transforms/Scalar/DxilLoopUnroll.cpp

```diff
@@ -810,6 +940,15 @@ bool DxilLoopUnroll::runOnLoop(Loop *L, LPPassManager &LPM) {
   }
 
   if (Succeeded) {
+    // We are going to try to clean up these allocas.
+    // But only the allocas that are not among the 
+    // blocks we are about to delete.
+    SmallVector<AllocaInst *, 8> CleanUpAllocas;
+    for (AllocaInst *AI : ProblemAllocas) {
+      if (!ToBeCloned.count(AI->getParent()))
```

Comment:
Shouldn't all alloca instructions be in the entry block?  And isn't the entry block one that we couldn't be cloning?